File: String-Construction.cpp
#include <bits/stdc++.h>

using namespace std;

string ltrim(const string &);
string rtrim(const string &);

/*
 * Complete the 'stringConstruction' function below.
 *
 * The function is expected to return an INTEGER.
 * The function accepts STRING s as parameter.
 */

int stringConstruction(string s)
{
    unordered_set<char> uniqueChars;
    
    for (char c : s)
    {
        uniqueChars.insert(c);
    }
    
    return uniqueChars.size();
}


int stringConstruction2(string s)
{
    set<char> uniqueChars;
    
    for (int i = 0; i < s.length(); ++i)
    {
        uniqueChars.insert(s[i]);
    }
    
    return uniqueChars.size();
}


int main()
{
    ofstream fout(getenv("OUTPUT_PATH"));

    string q_temp;
    getline(cin, q_temp);

    int q = stoi(ltrim(rtrim(q_temp)));

    for (int q_itr = 0; q_itr < q; q_itr++) {
        string s;
        getline(cin, s);

        int result = stringConstruction(s);

        fout << result << "\n";
    }

    fout.close();

    return 0;
}

string ltrim(const string &str)
{
    string s(str);
    s.erase(
        s.begin(), find_if(s.begin(), s.end(), [](unsigned char ch) { return !isspace(ch); })
    );
    return s;
}

string rtrim(const string &str)
{
    string s(str);
    s.erase(
        find_if(s.rbegin(), s.rend(), [](unsigned char ch) { return !isspace(ch); }).base(), s.end()
    );
    return s;
}

File: String-Construction.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;


class Result
{

    /*
     * Complete the 'stringConstruction' function below.
     *
     * The function is expected to return an INTEGER.
     * The function accepts STRING s as parameter.
     */

    public static int stringConstruction(string s)
    {
        HashSet<char> uniqueChars = new HashSet<char>();

        foreach (char c in s)
        {
            uniqueChars.Add(c);
        }

        return uniqueChars.Count;
    }    


    public static int stringConstruction2(string s)
    {
        SortedSet<char> uniqueChars = new SortedSet<char>();

        foreach (char c in s)
        {
            uniqueChars.Add(c);
        }

        return uniqueChars.Count;
    }
    
}
